pkg/envoy/lds/response.go

```diff
@@ -44,6 +33,10 @@ func NewResponse(meshCatalog catalog.MeshCataloger, proxy *envoy.Proxy, _ *xds_d
 
 	lb := newListenerBuilder(meshCatalog, svcAccount.ToServiceIdentity(), cfg, statsHeaders)
 
+	if proxy.Kind() == envoy.KindGateway {
```

Comment:
lets add the multicluster feature flag enabled check here instead please

Reply:
I had it here earlier, but @shashankram asked me to move it to the current location

Reply:
@snehachhabria this check is moved inside the function that gets called, so that ingress code can be integrated without needing to move the check back to where it is after the changes mode. This looks good to me.